### PR TITLE
Issue 57

### PR DESCRIPTION
**Pull request by motabass**

### CODE REVIEW

**Comment by motabass** [comment]
Default color is not working like before..because i was not sure what to choose...i may put it back to red...seems better than black..as it is right now. Empty color should still work but is not removing inputs anymore. I also did not check any tests yet🙈

**Comment by motabass** [comment]
I did some important fixes to my changes. It should now bahve like before. But i have still ExpressionChangedAfterChecked-Errors in tests. Need some more invstigation..

**Comment by tiaguinho** [comment]
@motabass remove the "WIP" from the title when this is ready for review.

Thank you!

**Comment by motabass** [comment]
Finally all Test are running successful and i even did some more improvements and refactorings.

**Comment by motabass** [comment]
https://github.com/tiaguinho/material-community-components/issues/91

The text is now updateing on pressing the "confirm" button.

File: angular.json [hunk 1]
```diff
@@ -193,7 +193,10 @@
       "style": "scss"
     },
     "@schematics/angular:directive": {
-      "prefix": "app"
+      "prefix": "mcc"
```
Comment (motabass) [inline angular.json:196]: prevents codelyzer tslint errors

**Comment by tiaguinho** [comment]
@motabass there are two tests failing. Can you take a look?

**Comment by motabass** [comment]
I did some final interactive rebase and force pushed to have the correct author in my commits. This PR is done from my side.

File: src/lib/color-picker/color-picker.service.ts [hunk 2]
```diff
@@ -23,18 +23,16 @@ export class MccColorPickerService {
 
   /**
    * Add new color to used colors
-   * @param color string
+   * @param colorString string
    */
-  addColor(color: string): void {
-    if (!color || !isValidColor(color)) {
+  addColor(colorString: string): void {
+    if (!colorString) {
```
Comment (tiaguinho) [inline src/lib/color-picker/color-picker.service.ts:29]: The lack of validation is causing an error.
Run the project and take a look at demo app.
The last two colors of ```usedStart``` are invalid colors, but the collections try to render something.

The validation is important so invalid colors aren't added.

Comment (motabass) [reply src/lib/color-picker/color-picker.service.ts]: Done. Thanks for the review.

**Comment by motabass** [comment]
By the way...i'm having a problem building the package with `ng-packagr -p src/lib/package.json`. I get the following error:
`ERROR: error TS6053: File 'C:/Users/marku/IdeaProjects/material-community-components/src/lib/index.ts' not found.
src/lib/material-community-components.ts:5:15 - error TS2307: Cannot find module './index' or its corresponding type declarations.

5 export * from './index';
`

**Comment by motabass** [comment]
Hi. there is 3 more commits fixing 2 small bugs and implementing #138

**Comment by tiaguinho** [comment]
@motabass not at all. In fact, I'm very happy with all the work you have put on this PR.
I'll take a look at this today.

Thank very much for all the contributions.

File: src/demo-app/app/color-picker/components/color-picker-alpha.component.html [hunk 3]
```diff
@@ -51,3 +51,26 @@
...
+      <mat-form-field>
+        <input matInput mccColorPickerOrigin #trigger2="mccColorPickerOrigin" formControlName="color" />
+      </mat-form-field>
+
+      <button type="submit" mat-raised-button color="primary">Submit
+        <small>(look the console)</small>
+      </button>
+    </form>
+  </mat-card-content>
+</mat-card>
```
Comment (tiaguinho) [inline src/demo-app/app/color-picker/components/color-picker-alpha.component.html:75]: The submit button isn't logging the current color on the console.

File: src/lib/color-picker/color-picker-selector.component.ts [hunk 4]
```diff
@@ -16,19 +16,20 @@ import {
...
-  value: number;
-}
 
-interface Offsets {
+interface Coordinates {
   x: number;
   y: number;
 }
 
+const INITIAL_COLOR: Instance = tinycolor({r: 255, g: 0, b: 0, a: 1});
```
Comment (tiaguinho) [inline src/lib/color-picker/color-picker-selector.component.ts:31]: The red color as default seems to be too aggressive.
I'm thinking that start with white will be a better fit.

File: src/lib/color-picker/color-picker-selector.component.ts [hunk 5]
```diff
@@ -427,179 +604,235 @@ export class MccColorPickerSelectorComponent
...
+    // this._blockContext.fillStyle = grdLeft;
+    // this._blockContext.fillRect(0, 0, this._bc.nativeElement.width, this._bc.nativeElement.height);
+    //
+    // const grdTop = this._hueSelectorContext.createLinearGradient(0, 0, 0, this._bc.nativeElement.height);
+    // grdTop.addColorStop(0, 'hsl(0, 0%, 100%)');
+    // grdTop.addColorStop(0.5, 'hsla(0, 0%, 100%, 0)');
+    // grdTop.addColorStop(0.5, 'hsla(0, 0%, 0%, 0)');
+    // grdTop.addColorStop(1, 'hsl(0, 0%, 0%)');
+    // this._blockContext.fillStyle = grdTop;
+    // this._blockContext.fillRect(0, 0, this._bc.nativeElement.width, this._bc.nativeElement.height);
```
Comment (tiaguinho) [inline src/lib/color-picker/color-picker-selector.component.ts:642]: Please remove this commented code from this PR.

Comment (motabass) [reply src/lib/color-picker/color-picker-selector.component.ts:642]: I pushed your requested changes

**Pull request merged**